Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -162,6 +162,46 @@ public override bool IsPointerOverGameObject(int pointerOrTouchId)
             return m_PointerStates[stateIndex].eventData.pointerEnter != null;
         }
 
+        /// <summary>
+        /// Returns the most recent raycast information for a given pointer or touch.
+        /// </summary>
+        /// <param name="pointerOrTouchId">ID of the pointer or touch. Meaning this should correspond to either
+        /// <c>PointerEventData.pointerId</c> or <see cref="ExtendedPointerEventData.touchId"/>. The pointer ID
+        /// generally corresponds to the <see cref="InputDevice.deviceId"/> of the pointer device. An exception
+        /// to this are touches as a <see cref="Touchscreen"/> may have multiple pointers (one for each active
+        /// finger). For touch, you can use the <see cref="TouchControl.touchId"/> of the touch.
+        ///
+        /// Negative values will return an invalid <see cref="RaycastResult"/>.</param>
+        /// <returns>The most recent raycast information.</returns>
+        /// <remarks>
+        /// This method is for the most recent raycast, but depending on when it's called is not guaranteed to be for the current frame.
+        /// This method can be used to determine raycast distances and hit information for visualization.
+        /// <br />
+        /// Use <see cref="RaycastResult.isValid"/> to determine if pointer hit anything.
+        /// </remarks>
+        /// <seealso cref="ExtendedPointerEventData.touchId"/>
+        /// <seealso cref="InputDevice.deviceId"/>
+        public RaycastResult GetLastRaycastResult(int pointerOrTouchId)
+        {
+            var stateIndex = GetPointerStateIndexFor(pointerOrTouchId);
+
+            if (stateIndex == -1)
+            {
+                for (var i = 0; i < m_PointerStates.length; ++i)
```

Comment:
nitpick, can be ignored, this is leaking implementation details of `GetPointerStateIndexFor`, a better way would be to add something like `bool getLastKnownState = false` argument to `GetPointerStateIndexFor` and only left `GetLastRaycastResult` to return `pointerCurrentRaycast`

Reply:
I ended up going the other way on this (I wasn't sure what you meant by GetLastKnownState=false).  And I ended up making it so GetPointerStateIndexFor() would also work for Touch & Device Ids.  Both callers had the same secondary check, and it's a private function so I felt it was better to make the getter cover that case too.